contracts/ERC721A.sol

```diff
@@ -163,7 +163,7 @@ contract ERC721A is IERC721A {
      * @dev See {IERC721-balanceOf}.
      */
     function balanceOf(address owner) public view override returns (uint256) {
-        if (_addressToUint256(owner) == 0) revert BalanceQueryForZeroAddress();
+        if (_isZero(owner)) revert BalanceQueryForZeroAddress();
```

Comment:
Can help check if `_isZero(_addressToUint256(owner))` gives the same gas results?